front/pages/api/w/[wId]/assistant/agent_configurations/[aId]/scope.ts

```diff
@@ -86,25 +85,63 @@ async function handler(
           },
         });
       }
+
       if (
         assistant.scope !== "private" &&
         bodyValidation.right.scope === "private"
       ) {
-        return apiError(req, res, {
-          status_code: 400,
-          api_error: {
-            type: "invalid_request_error",
-            message: "Non-private assistants cannot be set back to private.",
-          },
+        // switching an assistant back to private: the caller must be a user
+        if (!auth.user()) {
+          return apiError(req, res, {
+            status_code: 404,
+            api_error: {
+              type: "app_auth_error",
+              message:
+                "An assistant can only be set to private by an existing user of the workspace.",
+            },
+          });
+        }
+
+        if (assistant.scope === "workspace" && !auth.isBuilder()) {
+          return apiError(req, res, {
+            status_code: 404,
+            api_error: {
+              type: "app_auth_error",
+              message: "Only builders can modify workspace assistants.",
+            },
+          });
+        }
+
+        // ensure the assistant is not in the list of the user otherwise
+        // switching it back to private will make it disappear
+        const setRes = await setAgentUserListStatus({
+          auth,
+          agentId: assistant.sId,
+          listStatus: "in-list",
         });
+
+        if (setRes.isErr()) {
+          return apiError(req, res, {
+            status_code: 500,
+            api_error: {
+              type: "internal_server_error",
+              message: setRes.error.message,
+            },
+          });
+        }
       }
 
-      const result = await setAgentScope(
+      const result = await createOrUpgradeAgentConfiguration(
```

Comment:
We def don't want to create a new version for that do we?

Reply:
As per IRL, I understand the reasoning behind the change and I am OK with it.
We could argue that the "scope" isn't really part of an agent configuration and is more a "metadata" so creating an entire new version is def a bit heavy, but consistency with assistant builder is a strong argument here 👍